### PR TITLE
Feat/self password reset

### PR DESCRIPTION
**Pull request by Moro-Code**
# Summary | Résumé

Apologies for the large PR !

This PR allows users with cognito accounts to reset their passwords. 

There are two main flows to the password reset. 

1. A user which has an account with the status CONFIRMED in cognito ( meaning they have verified their email ) 

https://user-images.githubusercontent.com/26819992/206301187-e9ca2fed-021c-4abd-8391-17c2e289e140.mov

2. A user which has an account but has not confirmed their account 

https://user-images.githubusercontent.com/26819992/206301273-929b1359-40b5-4437-a038-4cd2b1466939.mov

If a user does not have an account. They will be redirected to the registration page

This PR also introduces the feature flag `passwordReset` . This is disabled by default. When disabled the Reset password link on the login page will not appear and users on the password reset page will be redirected to the login page

# Test instructions | Instructions pour tester la modification

Try out each flow and ensure you encounter no bugs

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet

The flow for an unconfirmed user is a bit crude. This can be refined in subsequent PRs. 

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [x] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [x] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [ ] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [x] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [x] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [ ] Have you modified the change log and updated any relevant documentation?
- [ ] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [x] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

File: lib/hooks/useAuth.ts [hunk 1]
```diff
@@ -157,47 +159,54 @@ export const useAuth = () => {
...
-            t("UsernameOrPasswordIncorrect.linkText"),
-            false
-          );
-        } else if (responseErrorMessage.includes("GoogleCredentialsExist")) {
-          await router.push("/admin/login");
-        } else {
-          setCognitoError(t("InternalServiceException"));
+    // try and sign the user in automatically if shouldSignIn is true, otherwise just
+    // call the passed in confirmationCallback
+    if (shouldSignIn) {
```
Comment (timarney) [inline lib/hooks/useAuth.ts:171]: Maybe use an early return here to get rid of some nesting of ifs

if (!shouldSignIn) {
confirmationCallback();
return 
}

Comment (Moro-Code) [reply lib/hooks/useAuth.ts:171]: Yup can do

File: pages/auth/resetpassword.tsx [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+
+  // validation schema for the initial form to send the forgot password verification code
+  const sendForgotPasswordValidationSchema = Yup.object().shape({
+    username: Yup.string()
+      .required(t("input-validation.required", { ns: "common" }))
+      .email(t("input-validation.email", { ns: "common" }))
+      .test(
+        "username-govEmail",
+        t("provideUsername.fields.username.error.validGovEmail"),
+        (value = "") => isValidGovEmail(value, emailDomainList.domains)
```
Comment (craigzour) [inline pages/auth/resetpassword.tsx:44]: When you will be rebasing on develop you will see that we do not need to pass any domains list anymore. The file has been removed so you will also have to remove the import at the top of the file

**Review by craigzour — APPROVED** [review]
LGTM! Will now test it locally

File: pages/auth/resetpassword.tsx (hunk 2 above)
Comment (Moro-Code) [reply pages/auth/resetpassword.tsx:44]: Sounds good will pull changes now and make this change

**Review by thiessenp-cds — APPROVED** [review]
Excellent work!!!

**Pull request merged**